plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/VectorField.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.correlation.core.index;
+
+import org.apache.lucene.document.Field;
+import org.apache.lucene.index.IndexableFieldType;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.plugin.correlation.core.index.serializer.CorrelationVectorSerializer;
+
+/**
+ * Generic Vector Field defining a correlation vector name, float array.
+ *
+ * @opensearch.internal
+ */
+public class VectorField extends Field {
+
+    /**
+     * Parameterized ctor for VectorField
+     * @param name name of the field
+     * @param value float array value for the field
+     * @param type type of the field
+     */
+    public VectorField(String name, float[] value, IndexableFieldType type) {
+        super(name, new BytesRef(), type);
+        try {
+            final byte[] floatToByte = CorrelationVectorSerializer.floatToByteArray(value);
+            this.setBytesValue(floatToByte);
+        } catch (Exception ex) {
```

Comment:
`CorrelationVectorSerializer.floatToByteArray` already catches IOExceptions and wraps them as an OpenSearchException. This class then catches the OpenSearchException and wraps it in a RuntimeException. Why?

I recommend that a wrap the IOException in a single place, _and_ that you do not bring in OpenSearchException and/or ExceptionHelpers and instead throw a base Java exception (maybe just RuntimeException) unless you have a good reason to do something else.

Reply:
throwing `java.lang.RuntimeException` only now.

Reply:
Don't catch `Exception` here. The risk is that if this code is refactored such that `floatToByteArray` where to throw another checked exception, this would swallow it which may or may not be appropriate. Namely, if a blocking call were added that threw an `InterruptedException` then this would code would swallow the interrupt without resetting the interrupt status. Just catch `IOException` here since that is the only checked exception being thrown.